posts/2021/04/pytorch-tensoriterator-internals-update.md

```diff
@@ -0,0 +1,337 @@
+<!--
+.. title: PyTorch TensorIterator Internals (TensorIteratorConfig Update)
+.. slug: pytorch-tensoriterator-internals-update
+.. date: 2021-04-09 08:00:00 UTC-06:00
+.. author: Kurt Mohler
+.. tags:
+.. category:
+.. link:
+.. description:
+.. type: text
+-->
+
+For contributors to the PyTorch codebase, one of the most commonly encountered
+C++ classes is
+[`TensorIterator`](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/TensorIterator.h).
+`TensorIterator` offers a standardized way to iterate over elements of
+a tensor, automatically parallelizing operations, while abstracting device and
+data type details.
+
+In April 2020, Sameer Deshmukh wrote a blog article discussing
+[PyTorch TensorIterator Internals](/blog/2020/04/pytorch-tensoriterator-internals/index.html). Recently,
+however, the interface has changed significantly. This post describes how to
+use the current interface as of January 2021. Much of the information from the
```

Comment:
just checking - still valid as of April 2021?

Reply:
Good catch, I changed it to April